lib/service/formula_wrapper.rb

```diff
@@ -42,6 +42,15 @@ def timed?
       @timed ||= formula.service.timed?
     end
 
+    # Delegate access to `formula.service.keep_alive?`.`
+    def keep_alive?
+      return unless @formula.service?
+
+      require_relative "../../../../../Homebrew/service"
```

Comment:
Could you split this into a `service` method that both `keepalive?` and `timed?` use?

Reply:
Like a `formula_wrapper#service` method that exposes the `formula#service` class? Yeah, we could do that but that would mean replacing `formula_wrapper#timed?` with `formula_wrapper#service? && formula_wrapper#service#timed?`. It would make the code a bit more verbose.

Reply:
A little bit, but it would also ensure we only have one ugly require statement.

Reply:
I see what you're saying. How about making a private `formula_wrapper#service` method that both `keep_alive?` and `timed?` can pass symbols to to call service methods? Then we'd be back to having just one ugly require statement.

Something like this:
```Ruby
# Delegate access to `formula.service.timed?`.
def timed?
  @timed ||= service(:timed?)
end

# Delegate access to `formula.service.keep_alive?`.`
def keep_alive?
  @keep_alive ||= service(:keep_alive?)
end

private

def service(method)
  return unless @formula.service?

  require_relative "../../../../../Homebrew/service"

  formula.service.public_send(method)
end
```

Reply:
I was thinking more like this (which might not be completely correct syntax since my Ruby is limited)

```
# Delegate access to `formula.service.timed?`.
def timed?
  @timed ||= service().timed?
end

# Delegate access to `formula.service.keep_alive?`.`
def keep_alive?
  @keep_alive ||= service().keep_alive?
end

private

def service
  return unless @formula.service?

  require_relative "../../../../../Homebrew/service"

  formula.service
end
```

Reply:
The problem with that is that #service could return nil so another check would be necessary in each method that calls it. We could also just leave the check out of #service entirely and do something like this but it's a bit more verbose.

```Ruby
# Delegate access to `formula.service.timed?`.
def timed?
  @timed ||= if service?
    service.timed?
  else
    false
  end
end

# Delegate access to `formula.service.keep_alive?`.`
def keep_alive?
  @keep_alive ||= if service?
    service.keep_alive?
  else
    false
  end
end

private

def service
 require_relative "../../../../../Homebrew/service"

  formula.service
end
```

Plus, looking at this code I don't know why we don't just validate the existence of a service when the FormulaWrapper gets instantiated for the first time. That would remove the need for all these #service? checks though it's not really related to this PR.

Reply:
Okay, well. I coded up something that looks similar to what I put in my last comment but I changed the name of the private method to #load_service to make it clear that it loads and returns the service.

Also, I assume the reason that this require_relative isn't at the top of the file has to do with name clashes or something like that but I could be reading that wrong.